PowerPointLabs/PowerPointLabs/ThisAddIn.cs

```diff
@@ -992,15 +992,22 @@ private void ThisAddInAfterNewPresentation(PowerPoint.Presentation pres)
             pres.Saved = Microsoft.Office.Core.MsoTriState.msoTrue;
         }
 
-        private void ThisAddInPrensentationOpen(PowerPoint.Presentation pres)
+        private void ThisAddInPresentationOpen(PowerPoint.Presentation pres)
         {
-            PowerPoint.DocumentWindow activeWindow = pres.Application.ActiveWindow;
-            string tempName = pres.Name.GetHashCode().ToString(CultureInfo.InvariantCulture);
-
-            // if we opened a new window, register the window with its name
-            if (!_documentHashcodeMapper.ContainsKey(activeWindow))
+            // Windows count could be zero if presentation is opened as preview of template slides
+            if (pres.Application.Windows.Count > 0)
             {
-                _documentHashcodeMapper[activeWindow] = tempName;
+                PowerPoint.DocumentWindow activeWindow = pres.Application.ActiveWindow;
+                string tempName = pres.Name.GetHashCode().ToString(CultureInfo.InvariantCulture);
+
+                // if we opened a new window, register the window with its name
+                if (!_documentHashcodeMapper.ContainsKey(activeWindow))
+                {
+                    _documentHashcodeMapper[activeWindow] = tempName;
+                }
+
+                // Refresh ribbon to grey out the menu / buttons if there are now at least one window

```

Comment:
I'm not sure if an actual presentation can start out with no active window but okay I think this can be done. Shall test it.

Reply:
Oh I did not want to enable the menu buttons if there is no active window so as to prevent the "no currently active document window" errors. Most of our PPTLabs features rely on an active window.